presto-jmx/src/main/java/io/prestosql/plugin/jmx/JmxMetadata.java

```diff
@@ -242,17 +254,14 @@ public ColumnMetadata getColumnMetadata(ConnectorSession session, ConnectorTable
     }
 
     @Override
-    public List<ConnectorTableLayoutResult> getTableLayouts(ConnectorSession session, ConnectorTableHandle table, Constraint constraint, Optional<Set<ColumnHandle>> desiredColumns)
+    public Optional<ConstraintApplicationResult<ConnectorTableHandle>> applyFilter(ConnectorSession session, ConnectorTableHandle handle, Constraint constraint)
     {
-        JmxTableHandle handle = (JmxTableHandle) table;
-        ConnectorTableLayout layout = new ConnectorTableLayout(new JmxTableLayoutHandle(handle, constraint.getSummary()));
-        return ImmutableList.of(new ConnectorTableLayoutResult(layout, constraint.getSummary()));
-    }
+        JmxTableHandle tableHandle = (JmxTableHandle) handle;
+        TupleDomain<ColumnHandle> predicate = constraint.getSummary();
 
-    @Override
-    public ConnectorTableLayout getTableLayout(ConnectorSession session, ConnectorTableLayoutHandle handle)
-    {
-        return new ConnectorTableLayout(handle);
+        JmxTableHandle newTableHandle = new JmxTableHandle(tableHandle.getTableName(), tableHandle.getObjectNames(), tableHandle.getColumnHandles(), tableHandle.isLiveData(), predicate);
+
+        return Optional.of(new ConstraintApplicationResult(newTableHandle, predicate));
```

Comment:
If the connector guarantees the results are going to satisfy the constraint, it's not necessary to return the predicate in the `ConstraintApplicationResult`. It can just indicate `TupleDomain.ALL`.